internal/batches/executor/executor.go

```diff
@@ -189,28 +190,32 @@ func (x *executor) do(ctx context.Context, task *Task, status taskStatusHandler)
 				status.CurrentlyExecuting = currentlyExecuting
 			})
 		},
+		// TODO: Why don't we pass the task to this?
```

Comment:
Good question!

Reply:
I'll try to address this in a follow-up!